java/src/com/mongodb/csfle/CSFLEHelpers.java

```diff
@@ -0,0 +1,219 @@
+                        .append("bsonType", bsonType)
+                        .append("algorithm",
+                                (isDeterministic) ? DETERMINISTIC_ENCRYPTION_TYPE : RANDOM_ENCRYPTION_TYPE));
+    }
+
+    private static Document createEncryptMetadataSchema(String keyId) {
+        List<Document> keyIds = new ArrayList<Document>();
+        keyIds.add(new Document()
+                .append("$binary", new Document()
+                        .append("base64", keyId)
+                        .append("subType", "04")));
+        return new Document().append("keyId", keyIds);
+    }
+
+    public static Document createJSONSchema(String keyId) throws IllegalArgumentException {
+        if (keyId.isEmpty()) {
+            throw new IllegalArgumentException("keyId must contain your base64 encryption key id.");
+        }
+        return new Document().append("bsonType", "object").append("encryptMetadata", createEncryptMetadataSchema(keyId))
+                .append("properties", new Document()
+                        .append("ssn", buildEncryptedField(keyId, "int", true))
+                        .append("bloodType", buildEncryptedField(keyId, "string", false))
+                        .append("medicalRecords", buildEncryptedField(keyId, "array", false))
+                        .append("insurance", new Document()
+                                .append("bsonType", "object")
+                                .append("properties",
+                                        new Document().append("policyNumber", buildEncryptedField(keyId, "int", true)))));
+    }
+
+    // Creates Normal Client
+    public static MongoClient createMongoClient(String connectionString) {
+        return MongoClients.create(connectionString);
+    }
+
+    // Creates KeyVault which allows you to create a key as well as encrypt and decrypt fields
+    public static ClientEncryption createKeyVault(String connectionString, String kmsProvider, byte[] localMasterKey, String keyVaultCollection) {
+        Map<String, Object> masterKeyMap = new HashMap<String, Object>();
+        masterKeyMap.put("key", localMasterKey);
+        Map<String, Map<String, Object>> kmsProviders = new HashMap<String, Map<String, Object>>();
+        kmsProviders.put(kmsProvider, masterKeyMap);
+
+        ClientEncryptionSettings clientEncryptionSettings = ClientEncryptionSettings.builder()
+                .keyVaultMongoClientSettings(MongoClientSettings.builder()
+                        .applyConnectionString(new ConnectionString(connectionString))
+                        .build())
+                .keyVaultNamespace(keyVaultCollection)
+                .kmsProviders(kmsProviders)
+                .build();
+
+        return ClientEncryptions.create(clientEncryptionSettings);
+    }
+
+    // You may need to update the following variable to point to your mongocryptd binary
+    private static String mongocryptdPath = "/usr/local/bin/mongocryptd";
```

Comment:
More typical (I hope) to just rely on mongocryptd being in your path

Reply:
This is tricky to make work for everyone since if you're using an IDE that does not have access to your PATH defined by your shell, it will not be able to find the binary. This scenario seems common and failed for me when I used Eclipse, started from the Finder (rather than a shell that might have my PATH defined). I think Eclipse may use `/etc/environment` for this purpose.

Reply:
Interesting.  I haven't had a problem in Intellij IDEA with this.

Reply:
I'd move this to the top then so it's more prominent.

Reply:
I will move it to the top. I wanted to keep the usage of the static variables near the methods that they would be used in, but maybe it becomes hidden when it's in the middle.